### PR TITLE
No name founded

### PR DESCRIPTION
**Pull request by Zangetsu38**
- Add gpu name on log info inside app init.
- fix startup project choice vita3k and no ALL BUILD with fix wrong path for project
- fix can allow change hardware flip without reboot emu before boot game.
- fix tab on setting dialog issue appear after update on 1.72 imgui. 
- fix missing change directory for format.sh
- Add on log backend render used
- fix vulkan render on gui working 
- fix command line for backend render

# Result: 
- GPU Name
image
- Startup project fixed by default.
image

### CODE REVIEW

**Comment by Zangetsu38** [comment]
Approve and merge if all is good 😍

File: vita3k/gui/src/settings_dialog.cpp [hunk 1]
```diff
@@ -103,9 +103,10 @@ static bool clear_and_refresh_game_list(HostState &host, GuiState &gui) {
 
 using namespace list;
 void draw_settings_dialog(GuiState &gui, HostState &host) {
+    const ImGuiWindowFlags settings_flags = ImGuiWindowFlags_AlwaysAutoResize;
     ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
-    ImGui::Begin("Settings", &gui.configuration_menu.settings_dialog, ImGuiWindowFlags_AlwaysAutoResize);
-    const auto settings_tab_flags = ImGuiTabBarFlags_None;
+    ImGui::Begin("Settings", &gui.configuration_menu.settings_dialog, settings_flags);
```
Comment (1whatleytay) [inline vita3k/gui/src/settings_dialog.cpp]: I'm not totally sure why you moved this out. If you say it's for a cleaner look, cool!

File: vita3k/interface.cpp [hunk 2]
```diff
@@ -202,6 +202,7 @@ static ExitCode load_app_impl(Ptr<const void> &entry_point, HostState &host, Gui
     LOG_INFO("Version: {}", host.game_version);
     LOG_INFO("Category: {}", game_category);
 
+    host.features.hardware_flip = host.cfg.hardware_flip;
```
Comment (1whatleytay) [inline vita3k/interface.cpp]: FeatureState moved here https://github.com/Vita3K/Vita3K/blob/master/vita3k/renderer/include/renderer/state.h#L16 :)

File: vita3k/gui/src/settings_dialog.cpp (hunk 1 above)
Comment (Zangetsu38) [reply vita3k/gui/src/settings_dialog.cpp]: based on admin works on demo imgui ^^

File: vita3k/interface.cpp (hunk 2 above)
Comment (Zangetsu38) [reply vita3k/interface.cpp]: ok thx :)

File: vita3k/config/src/config.cpp [hunk 3]
```diff
@@ -133,7 +133,7 @@ static ExitCode parse(Config &cfg, const fs::path &load_path, const std::string
     get_yaml_value(config_node, "color-surface-debug", &cfg.color_surface_debug, false);
     get_yaml_value(config_node, "hardware-flip", &cfg.hardware_flip, false);
     get_yaml_value(config_node, "performance-overlay", &cfg.performance_overlay, false);
-    get_yaml_value(config_node, "backend-renderer", &cfg.backend_renderer, std::string("OpenGL"));
+    get_yaml_value(config_node, "backend-renderer", &cfg.backend_renderer, static_cast<std::string>("OpenGL"));
```
Comment (1whatleytay) [inline vita3k/config/src/config.cpp]: Effectively, this casting is the same as `std::string()`. Use `std::string()` to make it consistent with the rest of the file.

File: vita3k/config/src/config.cpp [hunk 4]
```diff
@@ -188,7 +188,7 @@ ExitCode serialize_config(Config &cfg, const fs::path &output_path) {
     config_file_emit_single(emitter, "color-surface-debug", cfg.color_surface_debug);
     config_file_emit_single(emitter, "hardware-flip", cfg.hardware_flip);
     config_file_emit_single(emitter, "performance-overlay", cfg.performance_overlay);
-    config_file_emit_single(emitter, "backend-render", cfg.backend_renderer);
+    config_file_emit_single(emitter, "backend-renderer", cfg.backend_renderer);
```
Comment (1whatleytay) [inline vita3k/config/src/config.cpp]: Thank you! :D

File: vita3k/config/src/config.cpp [hunk 5]
```diff
@@ -289,7 +289,7 @@ ExitCode init_config(Config &cfg, int argc, char **argv, const Root &root_paths)
         po::options_description config_desc("Configuration");
         config_desc.add_options()
             ("archive-log,A", po::bool_switch(&command_line.archive_log), "Makes a duplicate of the log file with TITLE_ID and Game ID as title")
-            ("backend-renderer,B", po::value(&command_line.backend_renderer), "Renderer backend to use, either \"OpenGL\" or \"Vulkan\"")
+            ("backend-renderer,B", po::value<std::string>(&command_line.backend_renderer), "Renderer backend to use, either \"OpenGL\" or \"Vulkan\"")
```
Comment (1whatleytay) [inline vita3k/config/src/config.cpp]: This ought to be redundant too. Can you remove the `<std::string>`? C++ should add it for you.

File: vita3k/config/src/config.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/config/src/config.cpp]: i have set like this for get like other have static<int> ^^

File: vita3k/config/src/config.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/config/src/config.cpp]: ah, i have set it because other string here have this here ^^

**Comment by Zangetsu38** [comment]
> 
> 
> Cool! I think the FeatureState thing will give a build error, so fix that and it will be good.

What featurestate error ?

**Comment by 1whatleytay** [comment]
You have fixed it :D

**Comment by Zangetsu38** [comment]
ah lol

File: vita3k/config/src/config.cpp (hunk 3 above)
Comment (1whatleytay) [reply vita3k/config/src/config.cpp]: No need for strings, we have a constructor!

File: vita3k/config/src/config.cpp (hunk 5 above)
Comment (1whatleytay) [reply vita3k/config/src/config.cpp]: Okay 😄 should be good without it if you want to take the time to remove it.

**Comment by Zangetsu38** [comment]
> 
> 
> Nice! I have some feedback on the config selection thing. Can you just tackle that and I will approve.
> 
> I'm almost confident that it should work without [this](https://github.com/Vita3K/Vita3K/pull/577/files#diff-a6a8977717cb5a4e9908c4f9b0256552R255) and [this](https://github.com/Vita3K/Vita3K/pull/577/files#diff-a6a8977717cb5a4e9908c4f9b0256552R340) but I have given up.

for this, not can works without ^^, i have tested 50 different things during long time 3 hours lol

**Comment by 1whatleytay** [comment]
I'll trust you 😅

**Comment by Zangetsu38** [comment]
> 
> 
> You have fixed it :D



> 
> 
> I'll trust you 😅
for this, is command line, rhs is not config.h value get but comand line imput and so for change value is needed

File: vita3k/config/src/config.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/config/src/config.cpp]: forget finish line lol for get same other with static<int> so here string haha

File: vita3k/config/src/config.cpp (hunk 5 above)
Comment (Zangetsu38) [reply vita3k/config/src/config.cpp]: ah oki :D

File: vita3k/config/src/config.cpp (hunk 3 above)
Comment (Zangetsu38) [reply vita3k/config/src/config.cpp]: yes, i have just understund why other have static cast for int, stupid me lol

File: vita3k/config/src/config.cpp [hunk 6]
```diff
@@ -252,7 +252,7 @@ void merge_configs(Config &lhs, const Config &rhs, const std::string &new_pref_p
         lhs.hardware_flip = rhs.hardware_flip;
     if (lhs.performance_overlay != rhs.performance_overlay && (!init || rhs.performance_overlay))
         lhs.performance_overlay = rhs.performance_overlay;
-    if (lhs.backend_renderer != rhs.backend_renderer && (!init || !rhs.backend_renderer.empty()))
+    if (lhs.backend_renderer != rhs.backend_renderer && (!init || rhs.backend_renderer != std::string("OpenGL")))
```
Comment (Zangetsu38) [inline vita3k/config/src/config.cpp]: i have also remove static here lol

File: vita3k/config/src/config.cpp [hunk 7]
```diff
@@ -336,9 +336,8 @@ ExitCode init_config(Config &cfg, int argc, char **argv, const Root &root_paths)
             cfg.recompile_shader_path = std::move(command_line.recompile_shader_path);
             return QuitRequested;
         }
-        if (!var_map.count("backend-renderer")) {
-            command_line.backend_renderer = "";
-        }
+        if (var_map.count("backend-renderer"))
+            cfg.backend_renderer = command_line.backend_renderer;
```
Comment (Zangetsu38) [inline vita3k/config/src/config.cpp]: removed strd string here also, don't needed

**Comment by Zangetsu38** [comment]
@1whatleytay look is normaly full good here :P i learn how code works in same ,time lol

**Pull request merged**